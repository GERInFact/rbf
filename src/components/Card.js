import React from "react";
import "./card.css";

const Card = (props) => {
  const { id, name, email } = props;

  return (
    <div className="card">
      <div className="card_img-container">
        <img
          src={`https://robohash.org/${id}?200x200`}
          alt="A random robot face"
          className="img-container_img"
        />
      </div>
      <div className="card_info">
        <h3 className="info_name">{name || "Tester"}</h3>
        <p className="info_email">{email || "test@test.com"}</p>
      </div>
    </div>
  );
};

export default Card;
