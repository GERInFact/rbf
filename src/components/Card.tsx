import React from "react";
import "./card.css";
import {IUser} from "../containers/App";

const Card = ({id, name, email}: IUser) : JSX.Element => {
  return (
    <div className="card">
      <div className="card_img-container">
        <img
          src={`https://robohash.org/${id}?200x200`}
          alt="A random robot face"
          className="img-container_img"
        />
      </div>
      <div className="card_info">
        <h3 className="info_name">{name || "Tester"}</h3>
        <p className="info_email">{email || "test@test.com"}</p>
      </div>
    </div>
  );
};

export default Card;
