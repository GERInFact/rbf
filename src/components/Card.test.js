import React from "react";
import renderer from "react-test-renderer";
import Card from "./Card";

it("expect to render Card component", () => {
  const component = renderer.create(<Card />);
  expect(component.toJSON()).toMatchSnapshot();
});
