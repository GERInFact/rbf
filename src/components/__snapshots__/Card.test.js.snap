// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render Card component 1`] = `
<div
  className="card"
>
  <div
    className="card_img-container"
  >
    <img
      alt="A random robot face"
      className="img-container_img"
      src="https://robohash.org/undefined?200x200"
    />
  </div>
  <div
    className="card_info"
  >
    <h3
      className="info_name"
    >
      Tester
    </h3>
    <p
      className="info_email"
    >
      test@test.com
    </p>
  </div>
</div>
`;
